### PR TITLE
Add Ideal Gas Law for physics

### PR DESCRIPTION
**Pull request by kavienanj**
### Ideal Gas Law consists of 2 functions with return computed values for pressure and volume of a gas system according to given input values.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: physics/ideal_gas_law.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    example_2_moles = 0.5
+    example_2_kelvin = 273
+    examle_2_volume = volume_of_gas_system(
+        example_2_moles, example_2_kelvin, example_2_pressure
+    )
+    print(
+        f"""Volume(V) of a gas system with P = {example_2_pressure} [Pa],
+        n = {example_2_moles} [mol], T = {example_2_kelvin} [K]:"""
+    )
+    print(f"{examle_2_volume} [m^3]")
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:81]:
```suggestion
    print(f"{examle_2_volume} [m^3]")

```

File: physics/ideal_gas_law.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    332.57848
+    >>> pressure_of_gas_system(0.5, 273, 0.004)
+    283731.01575
+    """
+    if moles < 0 or kelvin < 0 or volume < 0:
+        raise Exception("Invalid inputs. Enter positive value.")
+    pressure: float = moles * kelvin * UNIVERSAL_GAS_CONSTANT / volume
+    return pressure
+
+
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:36]:
```suggestion
```

**Review by CaedenPH — APPROVED** [review]
Looks good

**Comment by kavienanj** [comment]
Added the suggestions! Please accept these changes

File: physics/ideal_gas_law.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+P	=	Pressure (Pa)
+V	=	Volume (m^3)
+n	=	Amount of substance (mol)
+R	=	Universal gas constant
+T	=	Absolute temperature (Kelvin)
+
+(Description adapted from https://en.wikipedia.org/wiki/Ideal_gas_law )
+"""
+
+UNIVERSAL_GAS_CONSTANT = 8.314462
```
Comment (ZeroDayOwl) [inline physics/ideal_gas_law.py:21]: I think you can add the units of R in a comment.

File: physics/ideal_gas_law.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    pressure: float = moles * kelvin * UNIVERSAL_GAS_CONSTANT / volume
+    return pressure
+
+
+def volume_of_gas_system(moles: float, kelvin: float, pressure: float) -> float:
+    """
+    >>> volume_of_gas_system(2, 100, 5)
+    332.57848
+    >>> volume_of_gas_system(0.5, 273, 0.004)
+    283731.01575
```
Comment (ZeroDayOwl) [inline physics/ideal_gas_law.py:42]: IMO you should add some failed cases in the doctest.

File: physics/ideal_gas_law.py (hunk 1 above)
Comment (ZeroDayOwl) [inline physics/ideal_gas_law.py:81]: You can exclude this part in the algorithm.

**Review by ZeroDayOwl — CHANGES_REQUESTED** [review]

File: physics/ideal_gas_law.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+def pressure_of_gas_system(moles: float, kelvin: float, volume: float) -> float:
+    """
+    >>> pressure_of_gas_system(2, 100, 5)
+    332.57848
+    >>> pressure_of_gas_system(0.5, 273, 0.004)
+    283731.01575
+    """
+    if moles < 0 or kelvin < 0 or volume < 0:
+        raise Exception("Invalid inputs. Enter positive value.")
+    pressure: float = moles * kelvin * UNIVERSAL_GAS_CONSTANT / volume
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:33]:
```suggestion
    pressure = moles * kelvin * UNIVERSAL_GAS_CONSTANT / volume
```

File: physics/ideal_gas_law.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+def volume_of_gas_system(moles: float, kelvin: float, pressure: float) -> float:
+    """
+    >>> volume_of_gas_system(2, 100, 5)
+    332.57848
+    >>> volume_of_gas_system(0.5, 273, 0.004)
+    283731.01575
+    """
+    if moles < 0 or kelvin < 0 or pressure < 0:
+        raise Exception("Invalid inputs. Enter positive value.")
+    volume: float = moles * kelvin * UNIVERSAL_GAS_CONSTANT / pressure
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:46]:
```suggestion
    volume = moles * kelvin * UNIVERSAL_GAS_CONSTANT / pressure
```

File: physics/ideal_gas_law.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    return volume
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:47]:
```suggestion
    return moles * kelvin * UNIVERSAL_GAS_CONSTANT / pressure
```

File: physics/ideal_gas_law.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+def pressure_of_gas_system(moles: float, kelvin: float, volume: float) -> float:
+    """
+    >>> pressure_of_gas_system(2, 100, 5)
+    332.57848
+    >>> pressure_of_gas_system(0.5, 273, 0.004)
+    283731.01575
+    """
+    if moles < 0 or kelvin < 0 or volume < 0:
+        raise Exception("Invalid inputs. Enter positive value.")
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:32]:
```suggestion
        raise ValueError("Invalid inputs. Enter positive value.")
```

File: physics/ideal_gas_law.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+def volume_of_gas_system(moles: float, kelvin: float, pressure: float) -> float:
+    """
+    >>> volume_of_gas_system(2, 100, 5)
+    332.57848
+    >>> volume_of_gas_system(0.5, 273, 0.004)
+    283731.01575
+    """
+    if moles < 0 or kelvin < 0 or pressure < 0:
+        raise Exception("Invalid inputs. Enter positive value.")
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:44]:
```suggestion
        raise ValueError("Invalid inputs. Enter positive value.")
```

File: physics/ideal_gas_law.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+UNIVERSAL_GAS_CONSTANT = 8.314462  # Unit - J mol-1 K-1
+
+
+def pressure_of_gas_system(moles: float, kelvin: float, volume: float) -> float:
+    """
+    >>> pressure_of_gas_system(2, 100, 5)
+    332.57848
+    >>> pressure_of_gas_system(0.5, 273, 0.004)
+    283731.01575
+    """
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:30]:
```suggestion
    >>> pressure_of_gas_system(3, -0.46, 23.5)
    Traceback (most recent call last):
        ...
    ValueError: Invalid inputs. Enter positive value.
    """
```

File: physics/ideal_gas_law.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    return moles * kelvin * UNIVERSAL_GAS_CONSTANT / volume
+
+
+def volume_of_gas_system(moles: float, kelvin: float, pressure: float) -> float:
+    """
+    >>> volume_of_gas_system(2, 100, 5)
+    332.57848
+    >>> volume_of_gas_system(0.5, 273, 0.004)
+    283731.01575
+    """
```
Comment (CaedenPH) [inline physics/ideal_gas_law.py:42]:
```suggestion
    >>> volume_of_gas_system(3, -0.46, 23.5)
    Traceback (most recent call last):
        ...
    ValueError: Invalid inputs. Enter positive value.
    """
```

**Review by CaedenPH — COMMENTED** [review]
Last things from me

**Comment by CaedenPH** [comment]
@cclauss I think this is worthy of review - Perhaps committing my changes might be good, however the pr in general looks good

**Review by cclauss — APPROVED** [review]

**Pull request merged**